### PR TITLE
enh(Poller Wizard): refacto - poller Wizard

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
## Description

I have completed the create poller server part.
- I deleted form-redux
- added Jotai atom
- useState to get data from form

because I can't test Post request, don't know if the functionality is as it should be.

# Original Design
## Remote Server
image
## Poller 
image
## Wizard installing page
image

# New Design
# select server type :
image

# Poller 
###  Create new Poller
image

###  Select a Poller
image

## Add advanced server configuration
image

# Remote Server
###  Create new Remote Server
image

## Add advanced server configuration
image

# Step Final (need design remake)
image

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by Yassir-BenBOUBKER** [comment]
I will start working on validation

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 1]
```diff
@@ -1,31 +1,86 @@
...
+      }
+    }
+  }, [stepOneFormData.server_ip]);
+
+  React.useEffect(() => {
+    if (waitList && !initialized) {
+      initializeFromRest(waitList.length > 0);
+    }
+    setInitialized(true);
+  }, [waitList]);
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:82]: We use to move useEffect hooks after component's functions declaration

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 2]
```diff
@@ -1,31 +1,86 @@
...
+        });
+      }
+    }
+  }, [stepOneFormData.server_ip]);
+
+  React.useEffect(() => {
+    if (waitList && !initialized) {
+      initializeFromRest(waitList.length > 0);
+    }
+    setInitialized(true);
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:81]: `initializeFromRest` already set `initialized` state to `true`

Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:81]:
```suggestion
    if (isNil(waitList) || initialized) {
      return;
    }
    initializeFromRest(waitList.length > 0);
```

File: www/front_src/src/PollerWizard/pollerStep2/index.tsx [hunk 3]
```diff
@@ -1,41 +1,83 @@
...
     success: boolean;
   }>({
     request: postData,
   });
+  const pollerData = useAtomValue<PollerData>(pollerAtom);
+  const setWizard = useUpdateAtom(setWizardDerivedAtom);
+
+  React.useEffect(() => {
+    getPollers();
+  }, []);
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep2/index.tsx:80]: Same comment as above, move the useEffect right above the returned JSX

**Review by Thebarda — COMMENTED** [review]
Overall, it's a great job 👍

**Comment by Yassir-BenBOUBKER** [comment]
# Validation

I make the Next Button disabled if one input is vide
image

the input will give an error if you select it, but don't type in it Or if delete all the text in it
image


all is good
image

**Review by Jordan91 — APPROVED** [review]
Nice :)

**Comment by Yassir-BenBOUBKER** [comment]
# remoteServerStep41 form validation
image

# change the design of Wizard Installing page
image

File: www/front_src/src/components/wizardFormInstallingStatus/index.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  statusCreating,
+  statusGenerating,
+  error,
+}: Props): JSX.Element => {
+  const classes = useStylesWithProps({ statusCreating, statusGenerating });
+
+  const { t } = useTranslation();
+  const loading = statusCreating === null || statusGenerating === null;
+  const hasError =
+    (statusCreating === false || statusGenerating === false) && error;
```
Comment (Thebarda) [inline www/front_src/src/components/wizardFormInstallingStatus/index.tsx:35]:
```suggestion
  const loading = isNil(statusCreating) || sNil(statusGenerating);
  const hasError =
    (equals(statusCreating, false) || equals(statusGenerating, false)) && error;
```

isNil and equals are imported from ramda https://ramdajs.com/docs/

File: www/front_src/src/components/wizardFormInstallingStatus/index.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      // className={styles['form-inner']}
+      >
+        <div className={classes.formHeading}>
+          <Typography variant="h6">{formTitle}</Typography>
+        </div>
+        {/* display loader until tasks are finished or error is displayed */}
+        <p className={classes.formText}>
+          <ContentWithCircularLoading alignCenter loading={loading}>
+            <span className={classes.statusCreating}>
+              {statusCreating != null ? (
```
Comment (Thebarda) [inline www/front_src/src/components/wizardFormInstallingStatus/index.tsx:53]:
```suggestion
              {not(isNil(statusCreating)) ? (
```

File: www/front_src/src/components/wizardFormInstallingStatus/index.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+              ) : (
+                '...'
+              )}
+            </span>
+          </ContentWithCircularLoading>
+        </p>
+        <p className={classes.formText}>
+          <ContentWithCircularLoading alignCenter loading={loading}>
+            <span className={classes.statusGenerating}>
+              {statusGenerating != null ? (
```
Comment (Thebarda) [inline www/front_src/src/components/wizardFormInstallingStatus/index.tsx:71]:
```suggestion
              {not(isNil(statusGenerating)) ? (
```

File: www/front_src/src/PollerWizard/remoteServerStep3/index.tsx [hunk 7]
```diff
@@ -1,23 +1,35 @@
...
 import { postData, useRequest } from '@centreon/ui';
 
 import WizardFormInstallingStatus from '../../components/wizardFormInstallingStatus';
 import routeMap from '../../reactRoutes/routeMap';
-
-interface Props {
-  pollerData;
+import { remoteServerAtom } from '../PollerAtoms';
+
+interface RemoteServerData {
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/remoteServerStep3/index.tsx:14]: You can use the interface defined in PollerAtoms

File: www/front_src/src/PollerWizard/remoteServerStep2/index.tsx [hunk 8]
```diff
@@ -84,38 +117,55 @@ const FormRemoteServerStepTwo = ({
...
+      </div>
+      <form autoComplete="off" onSubmit={handleSubmit}>
+        {remoteServers && (
+          <MultiAutocompleteField
+            fullWidth
+            label={t('Select pollers to be attached to this new Remote Server')}
+            options={remoteServers.map((c) => ({
+              id: c.id,
+              name: c.name,
+            }))}
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/remoteServerStep2/index.tsx:146]: Extract this to a specific variable

File: www/front_src/src/PollerWizard/remoteServerStep2/index.tsx [hunk 9]
```diff
@@ -84,38 +117,55 @@ const FormRemoteServerStepTwo = ({
...
   return (
-    <Form
-      goToPreviousStep={goToPreviousStep}
-      pollers={remoteServers}
-      onSubmit={handleSubmit}
-    />
+    <div>
+      <div className={classes.formHeading}>
+        <Typography variant="h6">
+          {t('Add advanced server configuration')}
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/remoteServerStep2/index.tsx:135]: Extract labels to a `translatedLabels.ts` file

File: www/front_src/src/PollerWizard/PollerAtoms.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  server_type?: string;
+  submitStatus?: boolean;
+}
+
+interface RemoteServerData {
+  centreon_central_ip?: string;
+  centreon_folder?: string;
+  db_password?: string;
+  db_user?: string;
+  // linked_pollers?: Array<string>;
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/PollerAtoms.ts:19]: I assume this is used

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 11]
```diff
@@ -39,33 +76,196 @@ const FormPollerStepOne = ({
...
+          />
+        </div>
+        {inputTypeManual ? (
+          <div className={classes.form}>
+            <TextField
+              fullWidth
+              required
+              error={
+                error.server_name.length > 0 ? error.server_name : undefined
+              }
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:169]: It's better to extract the JSX into the body of the component to improve readability

File: www/front_src/src/PollerWizard/PollerAtoms.ts (hunk 10 above)
Comment (Yassir-BenBOUBKER) [reply www/front_src/src/PollerWizard/PollerAtoms.ts:19]: no it not used

Comment (Thebarda) [reply www/front_src/src/PollerWizard/PollerAtoms.ts:19]: At least, make it optional and let it undefined

File: www/front_src/src/PollerWizard/remoteServerStep2/index.tsx [hunk 12]
```diff
@@ -60,19 +79,26 @@ const FormRemoteServerStepTwo = ({
...
-  const handleSubmit = (data): void => {
-    const dataToPost = { ...data, ...pollerData };
+  const handleSubmit = (event): void => {
+    event.preventDefault();
+    const dataToPost = {
+      ...pollerData,
+      linked_pollers: linkedPollers.map(({ id }) => id),
+    };
     dataToPost.server_type = 'remote';
+    setIsLoading(true);
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/remoteServerStep2/index.tsx:101]: You can use the 'sending' property returned by useRequest directly

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 13]
```diff
@@ -114,6 +114,9 @@ const FormPollerStepOne = ({
 
   const waitListOption = waitList?.map((c) => ({ id: c.ip, name: c.ip }));
 
+  const hasError = (stateName): string | undefined =>
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:117]: The function name doesn't sound correct, as it isn't a check for the error existence, but rather a retrieve of it's value. getError would sound more accurate to me

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 14]
```diff
@@ -16575,3 +16575,59 @@ msgstr "Votre mot de passe expirera dans"
 
 msgid "Password has expired"
 msgstr "Votre mot de passe a expiré"
+
+# msgid "Export generation timeout"
+# msgstr "Temps d'exécution maximal d'une génération exportatrice"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16580]:
```suggestion
# msgstr "Durée d'expiration de la génération d'export"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -16575,3 +16575,59 @@ msgstr "Votre mot de passe expirera dans"
...
+# msgstr "Créer un nouveau Collecteur"
+
+# msgid "Select a Poller"
+# msgstr "Sélectionnez un Collecteur"
+
+# msgid "Select Pending Poller IP"
+# msgstr "Sélectionnez l'adresse IP du collecteur en attente"
+
+# msgid "Do not check SSL certificate validation"
+# msgstr "Ne cochez pas la validation du certificat SSL"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16595]:
```suggestion
# msgstr "Ne pas vérifier la validité du certificat SSL"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 16]
```diff
@@ -16575,3 +16575,59 @@ msgstr "Votre mot de passe expirera dans"
...
+
+# msgid "Types"
+# msgstr "Les types"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16598]:
```suggestion
# msgstr "Types"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 17]
```diff
@@ -16575,3 +16575,59 @@ msgstr "Votre mot de passe expirera dans"
...
+
+# msgid "Uninstalled"
+# msgstr "Désinstallé"
+
+# msgid "Outdated"
+# msgstr "Périmé"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16604]:
```suggestion
# msgstr "Expiré"
```

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 18]
```diff
@@ -39,33 +78,188 @@ const FormPollerStepOne = ({
...
+  const getError = (stateName): string | undefined =>
+    error[stateName].length > 0 ? error[stateName] : undefined;
+
+  const nextDisabled =
+    Object.values(stepOneFormData).some((x) => x === '') ||
+    Object.values(error).some((x) => x !== '');
+
   React.useEffect(() => {
     getWaitList();
   }, []);
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:127]:
```suggestion
  }, []);
  
```

File: www/front_src/src/PollerWizard/PollerAtoms.ts [hunk 19]
```diff
@@ -0,0 +1,40 @@
+import { atom } from 'jotai';
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/PollerAtoms.ts:1]: this type of file should start with a lower case

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 20]
```diff
@@ -1,31 +1,70 @@
...
-interface Props
-  extends Pick<WizardFormProps, 'goToNextStep' | 'goToPreviousStep'> {
-  setWizard: (pollerWizard) => Record<string, unknown>;
+interface StepOneFormData {
+  centreon_central_ip: string;
+  server_ip: string;
+  server_name: string;
 }
 
 const FormPollerStepOne = ({
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:38]:
```suggestion
const PollerWizardStepOne = ({
```

File: www/front_src/src/PollerWizard/pollerStep2/index.tsx [hunk 21]
```diff
@@ -1,55 +1,119 @@
...
   './api/internal.php?object=centreon_configuration_remote&action=getRemotesList';
 const wizardFormEndpoint =
   './api/internal.php?object=centreon_configuration_remote&action=linkCentreonRemoteServer';
-interface Props
-  extends Pick<WizardFormProps, 'goToNextStep' | 'goToPreviousStep'> {
-  pollerData: Record<string, unknown>;
-  setWizard: (data) => void;
-}
 
+interface stepTwoFormData {
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep2/index.tsx:42]: interfaces should start with an Uppercase
```suggestion
interface StepTwoFormData {
```

File: www/front_src/src/PollerWizard/pollerStep2/index.tsx [hunk 22]
```diff
@@ -1,55 +1,119 @@
...
+  linked_remote_master: string;
+  linked_remote_slaves: Array<SelectEntry>;
+  open_broker_flow: boolean;
+}
 const FormPollerStepTwo = ({
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep2/index.tsx:47]:
```suggestion
const PollerWizardStepTwo = ({
```

File: www/front_src/src/PollerWizard/pollerStep3/index.tsx [hunk 23]
```diff
@@ -1,34 +1,25 @@
...
-}
-
-const FormPollerStepThree = ({ pollerData }: Props): JSX.Element => {
+const FormPollerStepThree = (): JSX.Element => {
   const { t } = useTranslation();
 
+  const pollerData = useAtomValue<PollerData>(pollerAtom);
+
   return (
     <WizardFormInstallingStatus
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep3/index.tsx:16]:
```suggestion
    <WizardFormSetupStatus
```

File: www/front_src/src/PollerWizard/pollerStep3/index.tsx [hunk 24]
```diff
@@ -1,34 +1,25 @@
...
+import { pollerAtom, PollerData } from '../PollerAtoms';
 import WizardFormInstallingStatus from '../../components/wizardFormInstallingStatus';
+import { labelFinalStep } from '../translatedLabels';
 
-interface Props {
-  pollerData;
-}
-
-const FormPollerStepThree = ({ pollerData }: Props): JSX.Element => {
+const FormPollerStepThree = (): JSX.Element => {
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/pollerStep3/index.tsx:10]:
```suggestion
const PollerWizardStepThree = (): JSX.Element => {
```

File: www/front_src/src/PollerWizard/remoteServerStep2/index.tsx [hunk 25]
```diff
@@ -1,53 +1,70 @@
...
+  labelApply,
+} from '../translatedLabels';
+import { Props, PollerOrRemoteList } from '../models';
 
 const getRemoteServersEndpoint =
   './api/internal.php?object=centreon_configuration_remote&action=getRemotesList';
 const wizardFormEndpoint =
   './api/internal.php?object=centreon_configuration_remote&action=linkCentreonRemoteServer';
 
 const FormRemoteServerStepTwo = ({
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/remoteServerStep2/index.tsx:36]:
```suggestion
const RemoteServerWizardStepTwo = ({
```

File: www/front_src/src/PollerWizard/remoteServerStep3/index.tsx [hunk 26]
```diff
@@ -1,23 +1,24 @@
...
+import {
+  labelExportGenerationTimeout,
+  labelFinalStep,
+} from '../translatedLabels';
 
 const exportTaskEndpoint =
   'internal.php?object=centreon_task_service&action=getTaskStatus';
 
-const FormRemoteServerStepThree = ({ pollerData }: Props): JSX.Element => {
+const FormRemoteServerStepThree = (): JSX.Element => {
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/remoteServerStep3/index.tsx:21]:
```suggestion
const RemoteServerWizardStepThree = (): JSX.Element => {
```

File: www/front_src/src/PollerWizard/remoteServerStep3/index.tsx [hunk 27]
```diff
@@ -86,21 +88,11 @@ const FormRemoteServerStepThree = ({ pollerData }: Props): JSX.Element => {
   return (
     <WizardFormInstallingStatus
```
Comment (bdauria) [inline www/front_src/src/PollerWizard/remoteServerStep3/index.tsx:89]:
```suggestion
    <WizardFormSetupStatus
```

File: www/front_src/src/components/wizardFormInstallingStatus/index.tsx [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+
+  const loading = isNil(statusCreating) || isNil(statusGenerating);
+  const hasError =
+    (equals(statusCreating, false) || equals(statusGenerating, false)) && error;
+
+  return (
+    <Paper
+    // className={
+    //   classnames(styles['form-container'], styles.installation)
+    // }
```
Comment (bdauria) [inline www/front_src/src/components/wizardFormInstallingStatus/index.tsx:47]:
```suggestion
```

File: www/front_src/src/components/wizardFormInstallingStatus/index.tsx [hunk 29]
```diff
@@ -0,0 +1,102 @@
...
+    >
+      <div
+      // className={styles['form-inner']}
```
Comment (bdauria) [inline www/front_src/src/components/wizardFormInstallingStatus/index.tsx:50]:
```suggestion
```

File: www/front_src/src/components/wizardFormInstallingStatus/index.tsx [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+      >
+        <div className={classes.formHeading}>
+          <Typography variant="h6">{formTitle}</Typography>
+        </div>
+        {/* display loader until tasks are finished or error is displayed */}
```
Comment (bdauria) [inline www/front_src/src/components/wizardFormInstallingStatus/index.tsx:55]:
```suggestion
```

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 31]
```diff
@@ -39,33 +78,189 @@ const FormPollerStepOne = ({
...
+  };
+
+  const waitListOption = waitList?.map((c) => ({ id: c.ip, name: c.ip }));
+
+  const getError = (stateName): string | undefined =>
+    error[stateName].length > 0 ? error[stateName] : undefined;
+
+  const nextDisabled =
+    Object.values(stepOneFormData).some((x) => x === '') ||
+    Object.values(error).some((x) => x !== '');
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:123]: Extract them to specific variables

File: www/front_src/src/PollerWizard/pollerStep2/index.tsx [hunk 32]
```diff
@@ -1,55 +1,119 @@
...
-  const { sendRequest: postWizardFormRequest } = useRequest<{
+
+  const { sendRequest: postWizardFormRequest, sending: loading } = useRequest<{
     success: boolean;
   }>({
     request: postData,
   });
 
-  const navigate = useNavigate();
+  const pollerData = useAtomValue<PollerData>(pollerAtom);
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep2/index.tsx:76]:
```suggestion
  const pollerData = useAtomValue<PollerData | null>(pollerAtom);
```

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 33]
```diff
@@ -39,33 +78,189 @@ const FormPollerStepOne = ({
...
+          <Button
+            color="primary"
+            disabled={nextDisabled}
+            size="small"
+            type="submit"
+            variant="contained"
+          >
+            {t(labelNext)}
+          </Button>
+        </div>
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:260]: You can extract those previous, next and finish buttons into a component with his own logic.
You will be able to re-use it across poller wizard's components

File: www/front_src/src/PollerWizard/remoteServerStep2/index.tsx [hunk 34]
```diff
@@ -1,53 +1,70 @@
...
 
-  const navigate = useNavigate();
+  const pollerData = useAtomValue(remoteServerAtom);
+  const setWizard = useUpdateAtom(setRemoteServerWizardDerivedAtom);
 
-  const filterOutDefaultPoller = (itemArr): Record<string, unknown> => {
-    for (let i = 0; i < itemArr.items.length; i += 1) {
-      if (itemArr.items[i].id === '1') itemArr.items.splice(i, 1);
+  const filterOutDefaultPoller = (itemArr): Array<PollerOrRemoteList> => {
+    for (let i = 0; i < itemArr.length; i += 1) {
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/remoteServerStep2/index.tsx:66]: We prefer to use the `filter` (or `reject` from Ramda) function here

**Comment by Yassir-BenBOUBKER** [comment]
Merge Conflict Fixed 👍

File: www/front_src/src/PollerWizard/forms/wizardButtons.tsx [hunk 35]
```diff
@@ -0,0 +1,47 @@
...
+
+const WizardButtons = ({
+  goToPreviousStep,
+  loading,
+  type,
+}: Props): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+
+  const label = type === 'Next' ? t(labelNext) : t(labelApply);
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/forms/wizardButtons.tsx:24]:
```suggestion
  const label = equals(type, 'Next') ? t(labelNext) : t(labelApply);
```

File: www/front_src/src/PollerWizard/forms/wizardButtons.tsx [hunk 36]
```diff
@@ -0,0 +1,47 @@
...
+  const loadIcon =
+    type === 'Apply' && loading ? <CircularProgress size={15} /> : null;
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/forms/wizardButtons.tsx:26]:
```suggestion
    equals(type, 'Apply') && loading ? <CircularProgress size={15} /> : null;
```

File: www/front_src/src/PollerWizard/forms/wizardButtons.tsx [hunk 37]
```diff
@@ -0,0 +1,47 @@
...
+
+import { Button, CircularProgress } from '@mui/material';
+
+import { labelPrevious, labelNext, labelApply } from '../translatedLabels';
+import { useStyles } from '../../styles/partials/form/PollerWizardStyle';
+
+interface Props {
+  goToPreviousStep: () => void;
+  loading: boolean;
+  type: 'Next' | 'Apply';
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/forms/wizardButtons.tsx:13]: You can use an enum here

File: www/front_src/src/PollerWizard/pollerStep1/index.tsx [hunk 38]
```diff
@@ -39,33 +77,179 @@ const FormPollerStepOne = ({
...
+    setWizard(stepOneFormData);
+    goToNextStep();
+  };
+
+  const waitListOption = waitList?.map((c) => ({ id: c.ip, name: c.ip }));
+
+  const getError = (stateName): string | undefined =>
+    error[stateName].length > 0 ? error[stateName] : undefined;
+
+  const atLeastOneVide = Object.values(stepOneFormData).some((x) => x === '');
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep1/index.tsx:120]:
```suggestion
  const isFormNotCompleted = values(stepOneFormData).some((x) => x === '');
```

File: www/front_src/src/PollerWizard/forms/wizardButtons.tsx [hunk 39]
```diff
@@ -0,0 +1,47 @@
...
+    type === 'Apply' && loading ? <CircularProgress size={15} /> : null;
+
+  return (
+    <div className={classes.formButton}>
+      <Button size="small" onClick={goToPreviousStep}>
+        {t(labelPrevious)}
+      </Button>
+      <Button
+        color="primary"
+        disabled={loading}
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/forms/wizardButtons.tsx:35]: I would add a `disabled` prop for this one

File: www/front_src/src/PollerWizard/pollerStep2/index.tsx [hunk 40]
```diff
@@ -1,55 +1,112 @@
...
   const getPollers = (): void => {
     getPollersRequest({ data: null, endpoint: getPollersEndpoint }).then(
       setPollers,
     );
   };
 
-  React.useEffect(() => {
-    getPollers();
-  }, []);
+  const handleChange = (evt): void => {
```
Comment (Thebarda) [inline www/front_src/src/PollerWizard/pollerStep2/index.tsx:78]:
```suggestion
  const handleChange = (event): void => {
```

**Review by Thebarda — APPROVED** [review]

File: www/front_src/src/Main/useUser.ts [hunk 41]
```diff
@@ -58,7 +58,7 @@ const useUser = (
           isExportButtonEnabled,
           locale: locale || 'en',
           name,
-          themeMode: 'light' as ThemeMode,
+          themeMode: 'dark' as ThemeMode,
```
Comment (Thebarda) [inline www/front_src/src/Main/useUser.ts:61]:
```suggestion
          themeMode: 'light' as ThemeMode,
```

**Review by bdauria — APPROVED** [review]

**Pull request merged**